guardian/pinboard
# instant emails for group mentions

Run `yarn database-setup`, selecting the stage and then the last option (to effectively run `019-TriggerEmailLambdaAfterItemInsert.sql`) for

---

In #259 we added the `email-lambda` but with only a special case for central production mentions, then in #268 we made emails for claims clearer and in #266 we turned on emails for missed individual mentions. All with the lambda being invoked by a schedule (every 5 mins), which makes sense for individual mentions as these go out if they've remained unread for an hour (currently) but for the central prod mentions and subsequent claims this can be a little slow...

SO now **all group mentions** are emailed almost instantly, by adding a DB 'after insert' trigger (like we have for desktop/push notifications) to send the item id of items with group mentions to the `email-lambda` as soon as they're created. For that item ID it receives, the `email-lambda` then performs a similar DB lookup as before to build up all the info it needs for the email send (rather than building it in the DB trigger, as that's trickier and harder to maintain), before sending out. 

The subject and email copy is slightly adjusted for the group mention emails, since they're instant (and 'missed' language didn't make sense)...
![image](https://github.com/guardian/pinboard/assets/19289579/01fedcf8-86b2-4eb4-8bd1-ffc6b2ae47f6)
... note that we place the item id of the mention item in the subject as a ref, and then for any claims we place the `relatedItemId` in the subject as the ref (therefore the same subject as the original request email) so they end up threaded (i.e. claim email is a reply to the original)...
![image](https://github.com/guardian/pinboard/assets/19289579/c07d18e9-2daa-4b3a-ae43-4252bc7b46d9)

## How to test

With this branch deployed to `CODE` simply mention a group (that you're a member of) in any pinboard chat (making sure to click Yes on the modal asking whether it should be a request), see the email arrive promptly, then claim the item in pinboard and see a further email arrive as a **reply** stating that you've claimed the request.
